### PR TITLE
Make sure theme families are  array before attempting to merge them

### PR DESCRIPTION
**Pull request by vcanales**
Follow up to https://github.com/WordPress/gutenberg/pull/55981

### CODE REVIEW

File: lib/experimental/fonts-api/class-wp-fonts-resolver.php [hunk 1]
```diff
@@ -214,7 +214,7 @@ private static function get_settings() {
 
 			// Merge the variation settings with the global settings.
 			$settings['typography']['fontFamilies']['theme'] = array_merge(
-				$settings['typography']['fontFamilies']['theme'],
+				$settings['typography']['fontFamilies']['theme'] ?? array(),
```
Comment (anton-vlasenko) [inline lib/experimental/fonts-api/class-wp-fonts-resolver.php]: Excuse me, but I don't understand why this change is necessary.
If `$settings['typography']['fontFamilies']['theme']` is `null`, it will be set to an array here (around line 205):
```php
if ( ! isset( $settings['typography']['fontFamilies']['theme'] ) || ! is_array( $settings['typography']['fontFamilies']['theme'] ) ) {
    $settings['typography']['fontFamilies']['theme'] = array();
}
```

Comment (vcanales) [reply lib/experimental/fonts-api/class-wp-fonts-resolver.php]: My best guess is that it's needed because the line you mention depends on us being within the editor: 

https://github.com/WordPress/gutenberg/blob/787f8eade02c93389cf11f7090428f0a5a5b8ba3/lib/experimental/fonts-api/class-wp-fonts-resolver.php#L190

https://github.com/WordPress/gutenberg/blob/787f8eade02c93389cf11f7090428f0a5a5b8ba3/lib/experimental/fonts-api/class-wp-fonts-resolver.php#L200

https://github.com/WordPress/gutenberg/blob/787f8eade02c93389cf11f7090428f0a5a5b8ba3/lib/experimental/fonts-api/class-wp-fonts-resolver.php#L201

If I understand that logic correctly, the second time we go through this condition, `$set_theme_structure` could be `false`, hence `$settings['typography']['fontFamilies']['theme']` never getting initialized as an array. Another solution would be to move that initialization out of the conditional on `$set_theme_structure`.

Comment (anton-vlasenko) [reply lib/experimental/fonts-api/class-wp-fonts-resolver.php]: > `$settings['typography']['fontFamilies']['theme']` never getting initialized as an array.

Yes, `$set_theme_structure` is `false` on the second iteration of the `foreach` loop. 

However, I don't understand how this affects `$settings['typography']['fontFamilies']['theme']`.
`$settings['typography']['fontFamilies']['theme']` is set to an array during the first iteration of the loop and remains an array in subsequent iterations. `array_merge()` always returns an array value.

Comment (vcanales) [reply lib/experimental/fonts-api/class-wp-fonts-resolver.php]: https://github.com/WordPress/gutenberg/blob/787f8eade02c93389cf11f7090428f0a5a5b8ba3/lib/experimental/fonts-api/class-wp-fonts-resolver.php#L222

After `array_unique` on line 222, `$settings['typography']['fontFamilies']['theme']` is going `null`, and it doesn't get reset on the next iteration. It's probably going `null` because `array_unique` is getting rid of a duplicate `key` that contains the the `theme` array, because the function isn't really comparing the value? I don't think `array_unique` works well with multidimensional arrays.

Comment (arthur791004) [reply lib/experimental/fonts-api/class-wp-fonts-resolver.php]: Yes, the main issue is the `$settings['typography']['fontFamilies']['theme']` becomes `null` after `array_unique` as it removes the key with the duplicate value.

Hence, I proposed https://github.com/WordPress/gutenberg/pull/56067 to
1. Set the flag of `array_unique` to `SORT_REGULAR`. The default flag is `SORT_STRING` and it means that the elements are converted to a string to do the comparison. However, the value after the conversion becomes `Array` and it leads to `$settings['typography']['fontFamilies']['theme']` being `null` if the `$settings['typography']['fontFamilies']` contains multiple keys.
   ```
   $settings['typography']['fontFamilies'] = array(
     'default' => array(),
     'theme' => array(),
   );

   print_r( (string) $settings['typography']['fontFamilies']['default'] ); // Array
   print_r( (string) $settings['typography']['fontFamilies']['theme'] ); // Array
   ```
2. Reset the `$settings['typography']['fontFamilies']['theme']` after the `array_unique` call to avoid it becoming `null` in the next iteration. WDYT?

**Comment by vcanales** [comment]
Closing in favor of https://github.com/WordPress/gutenberg/pull/56067, which seems like a more robust solution.

File: lib/experimental/fonts-api/class-wp-fonts-resolver.php (hunk 1 above)
Comment (anton-vlasenko) [reply lib/experimental/fonts-api/class-wp-fonts-resolver.php]: @vcanales 

> After array_unique on line 222, $settings['typography']['fontFamilies']['theme'] is going null, 

I'm not able to reproduce this.

Can you provide the values of `$settings` and `$variations` at which this issue occurs? Please see this code snippet: https://3v4l.org/InOuS#v7.3.0

> After `array_unique` is applied on line 222, `$settings['typography']['fontFamilies']['theme']` becomes null and does not reset in the subsequent iteration.

Can you provide an example of such behavior? Specifically, what exact array should I pass to `array_unique()` in order to make `$settings['typography']['fontFamilies']['theme']` a `null` value?
